### PR TITLE
refactor: remove mocks from CliParametersAnalyzerTest

### LINKED ISSUE #591: CLI tests fail randomly on CI
**Summary**

We're seeing the `provideStorageDirectoryCliParameterWithoutSpecifyingUrlShouldReturnFalse()` test randomly fail on CI. For example:
https://github.com/MobilityData/gtfs-validator/runs/1685201080

```
org.mobilitydata.gtfsvalidator.cli.CliParametersAnalyzerTest > provideStorageDirectoryCliParameterWithoutSpecifyingUrlShouldReturnFalse FAILED
    org.mockito.exceptions.verification.WantedButNotInvoked at CliParametersAnalyzerTest.java:113
```

This passes for me locally in IntelliJ.

**What I expect**

CI should be predictable and always pass unless there are bugs in the code

**What I see**

CI fails seemingly randomly

### PR DESCRIPTION
**Pull request by lionel-nj**
closes #591 

**Summary:**

This PR provides support to:
- remove mocks from `CliParametersAnalyzerTest` 
- fix randomly failing tests on CI

**Expected behavior:** 

Tests should pass on CI.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- ~[ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: main/src/test/java/org/mobilitydata/gtfsvalidator/cli/CliParametersAnalyzerTest.java [hunk 1]
```diff
@@ -17,130 +17,123 @@
...
 import org.junit.runners.JUnit4;
-import org.mockito.ArgumentCaptor;
-import org.mockito.Mockito;
 
 @RunWith(JUnit4.class)
 public class CliParametersAnalyzerTest {
-  private Handler mockHandler = null;
-  private ArgumentCaptor<LogRecord> logRecordCaptor = null;
+  @Test
+  public void isValid_false() {
```
Comment (aababilov) [inline main/src/test/java/org/mobilitydata/gtfsvalidator/cli/CliParametersAnalyzerTest.java:29]: Please split that into separate tests for each scenario.

Remember that a failed assertThat interrupts your test. In your code, you test multiple scenarios, so the first failed scenario prevents the following ones and test output is incomplete.

Comment (aababilov) [reply main/src/test/java/org/mobilitydata/gtfsvalidator/cli/CliParametersAnalyzerTest.java:29]: Please avoid code duplication. Create a helper static method that would accept your String [] argv, validate them and return the result of isValid().

**Review by aababilov — CHANGES_REQUESTED** [review]

**Comment by lionel-nj** [comment]
Thanks for review @aababilov, PTAL.

**Review by aababilov — CHANGES_REQUESTED** [review]

**Review by aababilov — CHANGES_REQUESTED** [review]
Please also remove mockito from main/build.gradle.

File: main/src/test/java/org/mobilitydata/gtfsvalidator/cli/CliParametersAnalyzerTest.java [hunk 2]
```diff
@@ -17,151 +17,139 @@
...
-  //                + "provided at the same time: '--input' and '--url'");
-  //
-  //    //noinspection ResultOfMethodCallIgnored because object is mocked
-  //    verify(mockArguments, times(1)).getUrl();
-  //    //noinspection ResultOfMethodCallIgnored because object is mocked
-  //    verify(mockArguments, times(2)).getInput();
-  //    verifyNoMoreInteractions(mockArguments, mockHandler);
-  //  }
+  @Test
+  public void noUrlNoInput_isNotValid() {
```
[the remark this answers is not in the document]
Comment (lionel-nj) [reply main/src/test/java/org/mobilitydata/gtfsvalidator/cli/CliParametersAnalyzerTest.java]: Thanks for the heads-up @aababilov!

**Comment by lionel-nj** [comment]
Thanks for review @aababilov! I modified the tests following your suggestions. I only kept test using long names since as you mentioned previously: using the short names for the same scenarios is not relevant - given that `ArgumentsTest` already checks arguments initialization whether we're using short or long names.

**Review by aababilov — APPROVED** [review]
Thanks!

**Pull request merged**